### PR TITLE
Added DeepSTARR

### PR DESCRIPTION
**Pull request by bernardo-de-almeida**
Add the DeepSTARR model for DNA sequence prediction of enhancer activities.

Zenodo: https://doi.org/10.5281/zenodo.5502060
Code: https://github.com/bernardo-de-almeida/DeepSTARR/tree/main/DeepSTARR
Manuscript: https://www.biorxiv.org/content/10.1101/2021.10.05.463203

### CODE REVIEW

File: DeepSTARR/model.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - python=3.7
+      - h5py
+      - pip
+    pip:
+      - keras>=2.0.4
+      - tensorflow>=1.0
+
+schema:
+    inputs:
+        shape: (249,4)
```
Comment (haimasree) [inline DeepSTARR/model.yaml:41]: I took a quick look into the failing tests. I think changing ```shape: (249, 4)``` to ```shape: (249, 4, 1)``` will fix the test

File: DeepSTARR/model.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          email: bernardo.almeida94@gmail.com
+    doc: Model predicting the activities of developmental and housekeeping enhancers in Drosophila S2 cells
+    cite_as: https://doi.org/10.1101/2021.10.05.463203
+    trained_on: Developmental and housekeeping quantitative enhancer activity. Held-out second half of chromosome 2R.
+    license: MIT
+
+dependencies:
+    conda:
+      - python=3.7
+      - h5py
```
Comment (haimasree) [inline DeepSTARR/model.yaml:33]: This does not matter for now but for ensuring that the model continues to operate as expected in future it is important to pin the dependencies. 
I am providing the versions as resolved in CI. Please consider adding them here.
```- h5py=3.6.0```

File: DeepSTARR/model.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - pip
```
Comment (haimasree) [inline DeepSTARR/model.yaml:34]: ```- pip=21.2.2```

File: DeepSTARR/model.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    pip:
+      - keras>=2.0.4
```
Comment (haimasree) [inline DeepSTARR/model.yaml:36]: ```- keras==2.7.0```

File: DeepSTARR/model.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      - tensorflow>=1.0
```
Comment (haimasree) [inline DeepSTARR/model.yaml:37]:
```- tensorflow==2.7.0

**Review by haimasree — CHANGES_REQUESTED** [review]
Thanks for the pr. I have provided (untested) a fix for the failing test and have provided a few more suggestions.I will accept the pr as soon as the tests are fixed.

File: DeepSTARR/model.yaml (hunk 1 above)
Comment (bernardo-de-almeida) [reply DeepSTARR/model.yaml]: I changed the shape as suggested, although the input shape in the model is (249, 4). See if it fixes the test

**Comment by bernardo-de-almeida** [comment]
I have performed the suggested changes, can you test it now? Many thanks

**Comment by bernardo-de-almeida** [comment]
We have CentOS 7.9 and pyrle 0.0.33.
Before I added the versions I currently use in my environment, but I updated now for your suggested pinned versions since they were from a successfully resolved conda environment

**Comment by haimasree** [comment]
> We have CentOS 7.9 and pyrle 0.0.33. Before I added the versions I currently use in my environment, but I updated now for your suggested pinned versions since they were from a successfully resolved conda environment

Great! Thanks. The main test is passing now :). The other test is failing due to a bug in kipoi. I am working on a fix as we speak. I will merge this pr afterwards.

**Comment by bernardo-de-almeida** [comment]
Great, many thanks for your help!

**Pull request merged**